### PR TITLE
Feature/hide empty tabs

### LINKED ISSUE #1102: 'Documentation' tab is visible when no documentation available for an API backend
# Steps
1. Visit https://nightly.apinf.io  
2. Sign in as owner  
3. Add API backend 
4. Do not upload documentation  
5. Sign out  
6. Login as regular user (doesn't own API backend)  
7. Go to 'Catalog'  
8. Select the API backend added above and click on 'View Details'  

Expected Result: Documentation tab should not be visible when there is no content  
Actual Result: 'Documentation' tab is visible   
# Findings

 User is notified:  'No documentation link is available' / 'No swagger document is available'   
# Operating system:

Linux : Ubuntu 14.04.03 LTS (Trusty)
# Browser:
1. Mozilla Firefox 44.0
2. Google Chrome 48.0.2564.103
# Screenshot

docs_unavailable_shows_tab

### LINKED ISSUE #1539: Useless tabs are shown for end user
When end user goes to API catalogue in APInf, there's same tabs for every API -  (Details, Documentation, Metadata, Feedback, Backlog, Export and Settings), whether there's content on page or not.

For example backlog tab usually says "No backlog items found for this API.". Since only API owner can only modify this, the whole tab could be hidden in case there are not backlog for the API. 

For many APIs only couple tabs are valid, rest of the tabs says "go away, nothing to see here".

Hiding useless tabs form API consumer is a good option if no content can be added there.

--- Following content added by @bajiat 2017-02-13 ----

# Note
* API owner and admin can always see all API tabs regardless of whether there is any content on the pages or not.
* Regular users and anonymous users can never see Export, Settings and Proxy tabs.
  * regular user = logged in, not admin and not manager of current API
* Details tab (default tab) is always available for all users, when the API is public.
* For private APIs, only API owners and allowed users can see the API profile page.
* Feedback tab should always be visible.

# Definition of done
* [ ] If there is no content on the Backlog tab, anonymous and regular users cannot see the tab.
* [ ] If there is no content on the Metadata tab, anonymous and regular users cannot see the tab.
* [ ] If there is no content on the Documentation tab, anonymous and regular users cannot see the tab.

Related to #1102

### PR DESCRIPTION
**Pull request by nnn**
Closes #1539

### CODE REVIEW

File: apis/collection/helpers.js [hunk 1]
```diff
@@ -237,4 +239,43 @@ Apis.helpers({
 
     return documentation;
   },
+  backlogIsNotEmpty () {
```
Comment (brylie) [inline apis/collection/helpers.js]: This function name is misleading. In effect, the function is doing two things:

- checking if there are any backlog items
- checking if user can manage

So, a more general name would suffice here, such as `userShouldSeeBacklogTab`.

File: apis/collection/helpers.js [hunk 2]
```diff
@@ -237,4 +239,43 @@ Apis.helpers({
...
+    const backlog = ApiBacklogItems.findOne({ apiBackendId: apiId });
+
+    // Check if backlog exist or user can manage
+    // to hide/show corresponding tab
+    if (backlog || this.currentUserCanManage()) {
+      return true;
+    }
+    return false;
+  },
+  apiDocsIsNotEmpty () {
```
Comment (brylie) [inline apis/collection/helpers.js]: This function name is misleading. In effect, the function is doing two things:

- checking if there are any ApiDocs items
- checking if user can manage

So, a more general name would suffice here, such as `userShouldSeeApiDocab`.

File: apis/collection/helpers.js [hunk 3]
```diff
@@ -237,4 +239,43 @@ Apis.helpers({
...
+    const apiDocs = ApiDocs.findOne({ apiId });
+
+    // Check if backlog exist or user can manage
+    // to hide/show corresponding tab
+    if (apiDocs || this.currentUserCanManage()) {
+      return true;
+    }
+    return false;
+  },
+  apiMetadataIsNotEmpty () {
```
Comment (brylie) [inline apis/collection/helpers.js]: This function name is misleading. In effect, the function is doing two things:

- checking if there are any API Metadata
- checking if user can manage

So, a more general name would suffice here, such as `userShouldSeeApiMetadataTab`.

**Comment by brylie** [comment]
I just realized that the functions are attached to the collection, as collection helpers. Why do we use collection helpers instead of template helpers? What is the main purpose of the functions?

**Comment by brylie** [comment]
@nnn is this ready?

**Comment by nnn** [comment]
Removed ApiDocs tab related code.
Anonymous is not able to see documentation, even if it exist.
Also getting weird console errors: 

```
I20170330-12:22:26.139(3)? Exception in queued task: TypeError: this.cursor._getCollectionName is not a function
I20170330-12:22:26.139(3)?     at Publication._getCollectionName (packages/reywood_publish-composite.js:140:63)
I20170330-12:22:26.139(3)?     at Publication.publish (packages/reywood_publish-composite.js:81:31)
I20170330-12:22:26.140(3)?     at Publication.createChildPublication (packages/reywood_publish-composite.js:147:13)
I20170330-12:22:26.140(3)?     at Array.forEach (native)
I20170330-12:22:26.140(3)?     at Function._.each._.forEach (packages/underscore.js:139:11)
I20170330-12:22:26.140(3)?     at Publication._publishChildrenOf (packages/reywood_publish-composite.js:144:7)
I20170330-12:22:26.140(3)?     at Object.added (packages/reywood_publish-composite.js:96:22)
I20170330-12:22:26.140(3)?     at [object Object].observeChangesCallbacks.added (packages/minimongo/observe.js:153:28)
I20170330-12:22:26.140(3)?     at self.applyChange.added (packages/minimongo/observe.js:53:44)
I20170330-12:22:26.141(3)?     at packages/mongo/observe_multiplex.js:182:30
I20170330-12:22:26.141(3)?     at Array.forEach (native)
I20170330-12:22:26.141(3)?     at Function._.each._.forEach (packages/underscore.js:139:11)
I20170330-12:22:26.142(3)?     at Object.task (packages/mongo/observe_multiplex.js:176:9)
I20170330-12:22:26.142(3)?     at [object Object]._.extend._run (packages/meteor.js:807:18)
I20170330-12:22:26.142(3)?     at packages/meteor.js:785:14
```
```

I20170330-11:59:10.777(3)? Exception from sub organizationApiLinksByOrganizationSlug id jPSHt5J72a2KZPCh9 TypeError: Cannot read property '_id' of undefined
I20170330-11:59:10.778(3)?     at [object Object].<anonymous> (organization_apis/collection/server/publications.js:51:26)
I20170330-11:59:10.779(3)?     at packages/matb33_collection-hooks.js:307:21
I20170330-11:59:10.780(3)?     at [object Object]._.extend.withValue (packages/meteor.js:1122:17)
I20170330-11:59:10.781(3)?     at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)
I20170330-11:59:10.782(3)?     at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1737:12)
I20170330-11:59:10.782(3)?     at [object Object]._.extend._runHandler (packages/ddp-server/livedata_server.js:1035:17)
I20170330-11:59:10.783(3)?     at [object Object]._.extend._startSubscription (packages/ddp-server/livedata_server.js:853:9)
I20170330-11:59:10.783(3)?     at [object Object]._.extend.protocol_handlers.sub (packages/ddp-server/livedata_server.js:625:12)
I20170330-11:59:10.783(3)?     at packages/ddp-server/livedata_server.js:559:43
```

**Comment by nnn** [comment]
@brylie yep.

**Pull request merged**